### ISSUE TITLE
Recent Objects - Configured to remember folder, but displaying folder and document instead

### ISSUE THREAD

**Issue by gsteimer** [tsgrp/hpi#573]
See the screencam below.  Something changed in recent objects, because I didn't see this behavior a few months ago.  Basically, I have the Recent Objects dashlet configured to remember the user's last 5 folders.  However, when clicking on a recent folder, the folder appears with a document pre-loaded in the stage.  The expected behavior would be to just load the folder, which in this trac's case, should auto-load the View All Documents action.

![recent-objects-with-doc](https://cloud.githubusercontent.com/assets/1377228/6129056/0f68b74c-b101-11e4-9289-1abb809ab9dd.gif)

**Comment by dgrumieaux** [comment]
For whoever starts looking at this issue, I would check to see if the
stage's documentId observable is not getting reset properly upon entering
the stage.

**Comment by jrubins** [comment]
In HPI Revision #11589, an update was made to `recentobject.js` to allow for PSI support from the recent objects dashlet. This update had the side-effect of causing any click on a recent folder to have the last document in the context added to the stage URL.

**Comment by jrubins** [comment]
It seems to me there are two possible solutions:

1. In `router.js`, before adding the "recent object" object to the cookie, check to see if the configured type for the dashlet is a folder type, and if so, don't add the context `documentId` value.
1. In `recentobjects.js`, when a user clicks on a folder, construct the URL to the stage differently depending on the object type configured of recent objects to show.

**Comment by m-brouillette** [comment]
I made the changes @jrubins  mentioned.  I thought I had mentioned that this would be a side effect.  Also sorry for seeing this issue earlier.

**Comment by jrubins** [comment]
We decided that we do not want to support PSIs in the Recent Objects dashlet. I removed the update made by @m-brouillette since it's no longer needed in `recentobject.js`. In `router.js`, I added a statement that only will add a recent object to the user preferences `recentObjects` cookie if the container and document `objectId` are not the same (that's how we determine if the object is a PSI or not). 

HPI Revision #11732.
